feat(data-exploration): person and cohort tables

mariusandra opened the pull request:
## Problem

Follow up to https://github.com/PostHog/posthog/pull/13116

## Changes

Replaces persons and cohort tables with data exploration versions.

![2022-12-05 16 28 45](https://user-images.githubusercontent.com/53387/205676272-843962fa-4f1f-42b2-a3eb-a4d9f48002fc.gif)
![2022-12-05 16 29 05](https://user-images.githubusercontent.com/53387/205676344-e08d5e27-8a21-4d2f-a6a8-b853be1d27c9.gif)

I had to do some annoying refactors to get the delete button properly extracted, apologies for all the noise around this in the PR:

![2022-12-07 15 37 01](https://user-images.githubusercontent.com/53387/206207811-b7690424-b405-4118-88ed-9f40d72f2da8.gif)

I also cleaned up the storybook query stories somewhat.

## How did you test this code?

Clicking around in the interface. Hopefully soon with visual regression tests.

mariusandra committed:
add a big reload button

mariusandra committed:
Merge branch 'master' into events-table-parity

mariusandra committed:
reload button

mariusandra committed:
load next data

mariusandra committed:
rename "more" to "actions"

mariusandra committed:
new events that load

mariusandra committed:
refactor new/next loading

mariusandra committed:
bug

mariusandra committed:
autoload

mariusandra committed:
Merge branch 'master' into events-table-parity

mariusandra committed:
new scene with a flag

mariusandra committed:
use only string columns

mariusandra committed:
move around

mariusandra committed:
configurator

mariusandra committed:
refactor again

mariusandra committed:
column config

mariusandra committed:
actually select them

mariusandra committed:
actually select them

mariusandra committed:
hide on save

mariusandra committed:
Merge branch 'master' into events-table-parity

mariusandra committed:
make the old events table work with the new events format

mariusandra committed:
configure columns

mariusandra committed:
clean up, add test

mariusandra committed:
move away from antd

mariusandra committed:
less ants, move move

mariusandra committed:
more cleanup

mariusandra committed:
export

mariusandra committed:
show icon, store person properties

mariusandra committed:
delete old code

mariusandra committed:
add readme

mariusandra committed:
link properties

mariusandra committed:
make it look closer to the old page

mariusandra committed:
url

mariusandra committed:
show buffer warning

mariusandra committed:
rename

mariusandra committed:
autoload spinner, fix load order

mariusandra committed:
add automatic row highlighting

mariusandra committed:
scope down

mariusandra committed:
comment

mariusandra committed:
more comments

mariusandra committed:
add link to session player

mariusandra committed:
move defaults to logic

mariusandra committed:
enable autoload

mariusandra committed:
sanitize old columns

mariusandra committed:
cleanup

mariusandra committed:
refactor

mariusandra committed:
key cleanup

mariusandra committed:
kea localstorage with global key

mariusandra committed:
query editor new look

mariusandra committed:
empty datatable stories

mariusandra committed:
query edit modal

mariusandra committed:
inline editor

mariusandra committed:
times of plenty

mariusandra committed:
add stories

mariusandra committed:
add stories

mariusandra committed:
add more stories

mariusandra committed:
friendly hint

mariusandra committed:
update snapshot

mariusandra committed:
replace a few more events tables

mariusandra committed:
replace a few more events tables

mariusandra committed:
add custom columns, add web performance table

mariusandra committed:
add custom columns, add web performance table

mariusandra committed:
set query locally

mariusandra committed:
set query locally

mariusandra committed:
persons page

mariusandra committed:
persons page

mariusandra committed:
fix person id types

mariusandra committed:
fix person id types

mariusandra committed:
basic persons table

mariusandra committed:
persons search

mariusandra committed:
data table on persons scene

mariusandra committed:
default columns depend on source node kind

mariusandra committed:
add to storybook

mariusandra committed:
enable "load more" on persons

mariusandra committed:
copy distinct id

mariusandra committed:
inline editor for data nodes

mariusandra committed:
make all queries locally editable if no setQuery and no readOnly

mariusandra committed:
these are uuids

mariusandra committed:
rename "custom" to "context"

mariusandra committed:
Merge branch 'replace-event-tables' into person-data-tables

mariusandra committed:
Merge branch 'master' into person-data-tables

mariusandra committed:
refactor stories

mariusandra committed:
rename InlineEditor to InlineEditorButton to avoid confusion

mariusandra committed:
not in use

mariusandra committed:
fix wrong filters

mariusandra committed:
add more hints

mariusandra committed:
persons scene url

mariusandra committed:
extract person deletion modal

mariusandra commented:
Actually ready for a look!

thmsobrmlr commented on frontend/src/scenes/persons/PersonsScene.tsx:
```diff
@@ -0,0 +1,26 @@
+
+    return (
+        <>
+            <PersonPageHeader />
```
Can we remove this header from `Persons`?

thmsobrmlr commented on frontend/src/scenes/persons/PersonsScene.tsx:
```diff
@@ -0,0 +1,26 @@
+import { personsSceneLogic } from 'scenes/persons/personsSceneLogic'
+
+export const scene: SceneExport = {
+    component: PersonsScene,
```
Can we remove the scene export from `Persons`?

thmsobrmlr reviewed (changes requested):
Great! Only one real issue that I ran into: The delete button for persons in cohorts does not show up for me.

Besides that some UX details regarding exports on the persons page, and a few inline comments. Nothing that needs to change, just observations:
- The exported CSV for "selected columns" contains a `person.$delete` column, which is probably unnecessary.
- The exported CSV for "all columns" contains a `value_at_data_point` column, which is probably unnecessary.
- Not related to this PR, but we might be able to improve this with data exploration - The exported CSV contains columns `id` and `distinct_ids.0`, but I can't filter for these columns (unclear what property I should use).
- The export button does not reflect the number of selected columns this any more:
    
    Before
    <img width="187" alt="Screenshot 2022-12-08 at 12 00 58" src="https://user-images.githubusercontent.com/1851359/206430405-d7239799-b6ef-4593-8393-1d1b149fbff4.png">

    After
    <img width="145" alt="Screenshot 2022-12-08 at 12 00 54" src="https://user-images.githubusercontent.com/1851359/206430408-9cc31fc5-0fd8-42b5-9eb8-6e36724ec9e7.png">

mariusandra committed:
Merge branch 'master' into person-data-tables

mariusandra committed:
backport few fixes

mariusandra committed:
fix a few issues

mariusandra committed:
remove delete from export

mariusandra committed:
export limit

mariusandra committed:
show filter count

mariusandra committed:
few fixes

mariusandra commented on frontend/src/scenes/cohorts/CohortEdit.tsx:
```diff
@@ -193,7 +198,7 @@ export function CohortEdit({ id }: CohortLogicProps): JSX.Element {
                 )}
 
-                {!isNewCohort && (
+                {typeof cohort.id === 'number' && (
```
Unfortunately this is needed for typing below when passing the cohort id to the query :/

thmsobrmlr replied:
I think a comment would be great in that case :)

mariusandra commented:
Thanks for the review! Fixed all the feedback, I hope. Would love to get this in to help move along work on the two [stacked](https://github.com/PostHog/posthog/pull/13203) [PRs](https://github.com/PostHog/posthog/pull/13233) on top of this, plus the [event table regressions](https://github.com/PostHog/posthog/issues/13223)... 

> * The exported CSV for "selected columns" contains a `person.$delete` column, which is probably unnecessary.
Removed!

> * The exported CSV for "all columns" contains a `value_at_data_point` column, which is probably unnecessary.
This is there also on the current live persons table export. I propose we deal with it later 🙃 

> * Not related to this PR, but we might be able to improve this with data exploration - The exported CSV contains columns `id` and `distinct_ids.0`, but I can't filter for these columns (unclear what property I should use).

Hopefully HogQL will help here

> * The export button does not reflect the number of selected columns this any more:

Fixed!

mariusandra committed:
call live_events_columns what it is

thmsobrmlr reviewed (approved):
🚢  it!
